dotnet/packages/Microsoft.TeamsAI/Microsoft.TeamsAI/AI/Models/OpenAIModel.cs

```diff
@@ -185,7 +191,11 @@ public async Task<PromptResponse> CompletePromptAsync(ITurnContext turnContext,
                 };
             }
 
-            if (!_options.UseSystemMessages!.Value && prompt.Output.Count > 0 && prompt.Output[0].Role == ChatRole.System)
+            // Get the model to use.
+            string model = promptTemplate.Configuration.Completion.Model ?? _deploymentName;
+            bool isO1Model = model.StartsWith("o1-");
```

Comment:
do we know how streaming works with the o1 model?

Reply:
https://openai.com/index/introducing-openai-o1-preview/#:~:text=include%20function%20calling%2C-,streaming,-%2C%20support%20for%20system - nope it does not support that.